torch/_jit_internal.py

```diff
@@ -160,15 +160,36 @@ def __getattr__(self, key):
     return createResolutionCallbackFromEnv(env())
 
 
+def get_closure_class(clazz):
+    """
+    Get a dictionary of closed over variables from a class.
+
+    The closure of a class is defined as the union of closure of it's member functions.
+    """
+    captures = {}
+    captures[clazz.__name__] = clazz
+    functions = inspect.getmembers(clazz, predicate=inspect.isfunction)
+    for _, datadesc in inspect.getmembers(clazz, predicate=inspect.isdatadescriptor):
+        # Case of getter / setter marked with @property / @something.setter
+        # Fetch getter and setter
+        functions.extend(inspect.getmembers(datadesc, predicate=inspect.isfunction))
+    for _, member in functions:
+        captures.update(get_type_hint_captures(member))
+        captures.update(get_closure(member))
+    return captures
+
+
 def get_closure(fn):
     """
     Get a dictionary of closed over variables from a function
     """
     captures = {}
     captures.update(fn.__globals__)
-
     for index, captured_name in enumerate(fn.__code__.co_freevars):
-        captures[captured_name] = fn.__closure__[index].cell_contents
+        try:
```

Comment:
when we have ValueError?

Reply:
This happens when torch.jit.script is used as decorator and the a method of class refers the class by name. Added comments.

Reply:
I think it's better to use hasattr() to check existence of cell_contents, that way the intention is more clear and we can avoid accidentally capture unintended errors.

Reply:
hasattr returns True though... i.e. that field exists, it just throws when accessed before it's setup completely :(